src/services/utilities.ts

```diff
@@ -1358,4 +1358,115 @@ namespace ts {
             diagnostics: error ? concatenate(diagnostics, [error]) : diagnostics
         };
     }
-}
+
+    export function getMissingAbstractMembersInsertion(classDecl: ClassDeclaration, resolvedType: ResolvedType, checker: TypeChecker, newlineChar: string): string {
+        const classSymbol = checker.getSymbolOfNode(classDecl);
+        const missingMembers = filterMissingMembers(filterAbstract(filterNonPrivate(resolvedType.members)), classSymbol.members);
+        return getInsertionsForMembers(missingMembers, classDecl, checker, newlineChar);
+    }
+
+    /**
+     * Finds members of the resolved type that are missing in the class pointed to by class decl
+     * and generates source code for the missing members.
+     */
+    export function getMissingMembersInsertion(classDecl: ClassDeclaration, resolvedType: ResolvedType, checker: TypeChecker, newlineChar: string): string {
+        const classSymbol = checker.getSymbolOfNode(classDecl);
+        const missingMembers = filterMissingMembers(filterNonPrivate(resolvedType.members), classSymbol.members);
+        return getInsertionsForMembers(missingMembers, classDecl, checker, newlineChar);
+    }
+
+    /**
+     * Finds the symbols in source but not target, generating a new map with the differences.
+     */
+    function filterMissingMembers(sourceSymbols: Map<Symbol>, targetSymbols: Map<Symbol>): Map<Symbol> {
+        const result: Map<Symbol> = createMap<Symbol>();
+        outer:
+        for (const sourceName in sourceSymbols) {
+            for (const targetName in targetSymbols) {

```

Comment:
Why not just test if `targetSymbols[targetName]` is `undefined`? no need for this inner loop.